### PR TITLE
BREAKING std/uuid: rework v4 and v5 module

### PR DESCRIPTION
**Pull request by lucacasonato**
This commit updates README, adds better JSDoc, and updates
implementations to use new `crypto` features.

### CODE REVIEW

File: uuid/v5.ts [hunk 1]
```diff
@@ -1,68 +1,56 @@
...
+import { concat } from "../bytes/mod.ts";
 import { assert } from "../_util/assert.ts";
 
 const UUID_RE =
   /^[0-9a-f]{8}-[0-9a-f]{4}-[5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i;
 
 /**
- * Validates the UUID v5.
- * @param id UUID value.
+ * Validate that the passed UUID is a RFC4122 v4 UUID.
```
Comment (wperron) [inline uuid/v5.ts:10]:
```suggestion
 * Validate that the passed UUID is an RFC4122 v5 UUID.
```

File: uuid/v4.ts [hunk 2]
```diff
@@ -1,23 +1,31 @@
 // Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
-import { bytesToUuid } from "./_common.ts";
 
 const UUID_RE =
   /^[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i;
 
 /**
- * Validates the UUID v4.
- * @param id UUID value.
+ * Validate that the passed UUID is a RFC4122 v4 UUID.
```
Comment (wperron) [inline uuid/v4.ts:7]:
```suggestion
 * Validate that the passed UUID is an RFC4122 v4 UUID.
```

**Review by wperron — APPROVED** [review]
LGTM with just some minor changes

**Review by kt3k — APPROVED** [review]
LGTM

**Pull request merged**